nexus/src/app/silo.rs

```diff
@@ -97,6 +97,26 @@ impl super::Nexus {
         Ok(shared::Policy { role_assignments })
     }
 
+    pub async fn silo_fetch_policy_by_id(
+        &self,
+        opctx: &OpContext,
+        silo_id: Uuid,
+    ) -> LookupResult<shared::Policy<authz::SiloRole>> {
+        let (.., authz_silo) = LookupPath::new(opctx, &self.db_datastore)
+            .silo_id(silo_id)
```

Comment:
Feels a bit sad to copy `silo_fetch_policy` with only this line being different, may extract a private helper that takes the `authz_silo` instead.